wasm.md

```diff
@@ -1080,24 +1069,24 @@ Currently, only one memory may be accessible to a module, and thus the `<mAddr>`
 
 ```k
     syntax Defn       ::= MemoryDefn
-    syntax MemoryDefn ::= "(" "memory"                            ")"
-                        | "(" "memory"     Int                    ")" // Size only
-                        | "(" "memory"     Int Int                ")" // Min and max.
-                        | "(" "memory" "(" "data" DataStrings ")" ")"
-                        |     "memory" "{" Int MaxBound "}"
- // -------------------------------------------------------
-    rule <k> ( memory                 ) => memory { 0   .MaxBound } ... </k>
-    rule <k> ( memory MIN:Int         ) => memory { MIN .MaxBound } ... </k>
+    syntax MemoryDefn ::= "(" "memory" OptionalId Limits                     ")"
+                        | "(" "memory" OptionalId "(" "data" DataStrings ")" ")"
+                        |     "memory" "{" OptionalId Int MaxBound "}"
+ // ------------------------------------------------------------------
+    rule <k> ( memory ID:OptionalId MIN:Int         ) => memory { ID MIN .MaxBound } ... </k>
       requires MIN <=Int #maxMemorySize()
-    rule <k> ( memory MIN:Int MAX:Int ) => memory { MIN MAX       } ... </k>
+    rule <k> ( memory ID:OptionalId MIN:Int MAX:Int ) => memory { ID MIN MAX       } ... </k>
       requires MIN <=Int #maxMemorySize()
        andBool MAX <=Int #maxMemorySize()
-    rule <k> ( memory ( data DS ) )
-          =>  memory { #lengthDataPages(DS) #lengthDataPages(DS) }
+    rule <k> ( memory ( data DS ) ) => ( memory #freshId(NEXTID) (data DS) ) ... </k>
```

Comment:
Why do you still need this rule? Shouldn't this be handled as `ID:OptionalId `?

Reply:
It's needed for the desugaring in the next rule. It's also how it looks in the spec, on page 118 and 119, under the abbreviations for tables and memories.

Reply:
Okay.. It looks fine then